pkg/driver/common/operator/hooks.go

```diff
@@ -141,6 +141,40 @@ func getHostedControlPlaneTolerations(hostedControlPlaneLister hypev1beta1lister
 	return tolerations, nil
 }
 
+// withHyperShiftLabels sets Deployment labels on a HyperShift hosted control-plane.
+func withHyperShiftLabels(c *clients.Clients) (dc.DeploymentHookFunc, []factory.Informer) {
+	hook := func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		labels, err := getHostedControlLabels(
+			c.ControlPlaneHypeInformer.Hypershift().V1beta1().HostedControlPlanes().Lister(),
+			c.ControlPlaneNamespace)
+		if err != nil {
+			return err
+		}
+		deployment.Spec.Template.Labels = labels
```

Comment:
This will replace all existing labels. The new labels should be *added*, so the deployment's labelSelector still works. In addition, existing labels should not be replaced with new values, again because of the deployment's labelSelector.